File: 프로그래머스/Level2/다음 큰 숫자.py
# https://programmers.co.kr/learn/courses/30/lessons/12911

def solution(n):
    n_bin = bin(n)[2:]
    count1 = n_bin.count('1')
    answer = n + 1
    while True:
        temp_bin = bin(answer)[2:]
        if count1 == temp_bin.count('1'):
            break
        answer += 1
    return answer

File: 프로그래머스/Level2/다음 큰 숫자.js
// https://programmers.co.kr/learn/courses/30/lessons/12911

const two = (n) => {
	let t = '';
	while (n !== 0) {
		const res = n % 2;
		n = parseInt(n / 2);
		t = res + t;
	}
	return t.split('').filter((a) => a === '1').length;
};

const solution = (n) => {
	let temp = n + 1;
	while (true) {
		if (two(n) === two(temp)) return temp;
		temp++;
	}
};
